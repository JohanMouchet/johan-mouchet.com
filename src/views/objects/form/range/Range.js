// @flow

import * as React from "react";
import cx from "classnames";
import "./Range.scss";

type Props = {
  id?: string,
  name: string,
  className?: string | Array<string> | Object,
  attributes?: { [key: string]: any }
};

const Range = ({ id, name, className, attributes }: Props) => {
  const classNames = cx("o-form__range", className);

  return (
    <input
      id={id}
      type="range"
      name={name}
      className={classNames}
      {...attributes}
    />
  );
};

export default Range;
