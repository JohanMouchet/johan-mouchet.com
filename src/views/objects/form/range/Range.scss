@import "assets/styles/settings/_settings";
@import "assets/styles/tools/_tools";
@import "../_settings";

/* (Experimental, won't work in all browsers) */
.o-form__range {
  appearance: none;
  width: 100%;
  height: 2px;
  background-color: $s-color-gray;
  outline: none;
  margin-top: t-vertical-rhythm(0.5, "rem");
  margin-bottom: t-vertical-rhythm(0.5, "rem");
  padding: 0;
  transition: background 0.2s ease-out;

  @include t-wekkit-hdpi-small-dimension("height", 2);

  &:not(:disabled) {
    cursor: pointer;
  }

  // Firefox overrides
  &::-moz-range-track {
    background-color: $s-color-gray;
    border: 0;
  }

  /* stylelint-disable no-descending-specificity */
  &::-moz-focus-inner,
  &::-moz-focus-outer {
    border: 0;
  }

  &::-ms-track {
    height: auto;
    border-radius: t-vertical-rhythm(0.4, "rem");
  }
  /* stylelint-enable no-descending-specificity */

  /** Thumb **/
  @mixin range-thumb($vendor-prefix) {
    &#{$vendor-prefix} {
      appearance: none;
      width: t-vertical-rhythm(0.8, "rem");
      height: t-vertical-rhythm(0.8, "rem");
      border-radius: 50%;
      border: 2px solid $s-color-secondary;
      background: $s-color-white;
    }
  }

  @include range-thumb("::-webkit-slider-thumb");
  @include range-thumb("::-moz-range-thumb");

  /* States */

  /** Hover **/
  &:hover {
    background-color: t-map-deep-get($form-settings, hover, color);

    &::-moz-range-track {
      background-color: t-map-deep-get($form-settings, hover, color);
    }
  }

  /** Focus **/
  &:focus {
    background-color: t-map-deep-get($form-settings, focus, color);

    &::-moz-range-track {
      background-color: t-map-deep-get($form-settings, focus, color);
    }
  }

  /** Disabled **/
  &:disabled {
    background-color: t-map-deep-get($form-settings, disabled, color);

    &::-moz-range-track {
      background-color: t-map-deep-get($form-settings, disabled, color);
    }

    &::-webkit-slider-thumb {
      background-color: t-map-deep-get(
        $form-settings,
        disabled,
        color-secondary
      );
      border-color: t-map-deep-get($form-settings, disabled, color);
    }

    &::-moz-range-thumb {
      background-color: t-map-deep-get(
        $form-settings,
        disabled,
        color-secondary
      );
      border-color: t-map-deep-get($form-settings, disabled, color);
    }
  }
}
