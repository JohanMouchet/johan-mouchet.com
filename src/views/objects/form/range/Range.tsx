import React from "react";
import cx from "classnames";
import styles from "./Range.module.scss";

type Props = {
  id?: string;
  name: string;
  className?: string | string[] | { [key: string]: boolean };
  attributes?: boolean | number | string;
};

const Range: React.FC<Props> = ({ id, name, className, attributes }) => {
  const classNames = cx("o-form__range", className);

  return (
    <input
      id={id}
      type="range"
      name={name}
      className={classNames}
      {...attributes}
    />
  );
};

export default Range;
