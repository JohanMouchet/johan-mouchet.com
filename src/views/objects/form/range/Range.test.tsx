import React from "react";
import { render } from "@testing-library/react";
import Range from ".";

it("Renders <Range /> without crashing", () => {
  expect(render(<Range name="name" />).container.firstChild).toMatchSnapshot();
});
