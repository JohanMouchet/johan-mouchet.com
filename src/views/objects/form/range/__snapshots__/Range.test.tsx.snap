// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders <Range /> without crashing 1`] = `
<input
  className="o-form__range"
  name="name"
  type="range"
/>
`;
